File: services/api-gateway/src/main/java/com/cvassistant/apigateway/model/ChatMessage.java
package com.cvassistant.apigateway.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "chat_messages")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ChatMessage {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "session_id", nullable = false)
    private ChatSession session;

    @Column(nullable = false)
    private String role; // 'user' or 'assistant'

    @Column(columnDefinition = "TEXT", nullable = false)
    private String content;

    private LocalDateTime timestamp;

    @PrePersist
    protected void onCreate() {
        timestamp = LocalDateTime.now();
    }
}

File: services/api_gateway/CvAssistant.ApiGateway.Domain/Entities/ChatMessage.cs
namespace CvAssistant.ApiGateway.Domain.Entities;

public class ChatMessage
{
    public long Id { get; set; }
    public long SessionId { get; set; }
    public string Role { get; set; } = string.Empty; // 'user' or 'assistant'
    public string Content { get; set; } = string.Empty;
    public DateTime Timestamp { get; set; }

    // Navigation property
    public ChatSession? Session { get; set; }
}
